src/libs/duplicate.c

```diff
@@ -249,8 +251,12 @@ void gui_post_expose(dt_lib_module_t *self, cairo_t *cri, int32_t width, int32_t
 
   // if not cached, load or reload a mipmap
   int res = 0;
-  if(d->preview_id != d->imgid || d->preview_zoom != nz * zoom_ratio || !d->preview_surf)
+  if(d->preview_id != d->imgid || d->preview_zoom != nz * zoom_ratio || !d->preview_surf 
+      || d->preview_width != width || d->preview_height != height)
```

Comment:
you are comparing d->preview_width but this is never set before.

The two new fields preview_width & preview_height must be initialized to 0 before. Or maybe this is already the case ? g_malloc0() ?

Reply:
I can do this, but as it stands, the first time this function is run for an image, d->preview_surf will be NULL. So the additional statements in the 'if' won't get run and the values will be set to the input height and width.

Where should it be initialised? In gui_init()?

Reply:
Can't hurt so I've changed to initialise in gui_init